src/Connection/StreamConnection.php

```diff
@@ -163,8 +175,17 @@ private function read(int $length, string $errorMessage) : string
             return $data;
         }
 
+
+        $fullMessage = $errorMessage . '. Last error: ';
+        $err = error_get_last();
+        if ($err) {
+            $fullMessage = $fullMessage . "error type " . $err['type'] . ', message: ' . $err['message'];
+        } else {
+            $fullMessage = $fullMessage . 'error unavailable';
+        }
+
         /** @psalm-suppress PossiblyNullArgument */
         $meta = \stream_get_meta_data($this->stream);
-        throw new CommunicationFailed($meta['timed_out'] ? 'Read timed out' : $errorMessage);
+        throw new CommunicationFailed($meta['timed_out'] ? 'Read timed out' : $fullMessage);
```

Comment:
Why create a `$fullMessage` and throw it away on timeout error? Does `error_get_last()` return any useful information that can be added to the timeout error message? If not, let's call `error_get_last()` only when necessary.

Reply:
Done, Now I use factory method only if it is not a timeout.